### PR TITLE
Add language specification to property names

### PR DESCRIPTION
**Pull request by externl**
This is the first step in implementing #2746. Namely,

Properties in `PropertyNames.xml` now require an additional attribute `languages="..."`. This specifies the languages which support this property. Languages which do not support the property do not generate an entry for it. As a consequence  they'll warnings will be issued they you try to use an unsupported property. 

`languages` is a comma separated list of `cpp`, `csharp`, `java`, `js`, or a `all` (by itself).

Reworked `makeprops` to delay generation until after we've parsed all the properties. We need to know ahead of time if an entire section will be empty or not.

### CODE REVIEW

File: config/PropertyNames.xml [hunk 1]
```diff
@@ -195,363 +195,364 @@ generated from the section label.
...
-        <suffix name="Locator" class="proxy"/>
-        <suffix name="PublishedEndpoints" />
-        <suffix name="ReplicaGroupId" />
-        <suffix name="Router" class="proxy"/>
-        <suffix name="ProxyOptions" />
-        <suffix name="ThreadPool" class="threadpool" />
-        <suffix name="MessageSizeMax" />
+        <suffix name="AdapterId" languages="cpp,csharp,java" />
+        <suffix name="Connection" class="connection" languages="cpp,csharp,java"/>
+        <suffix name="Endpoints" languages="all" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:234]: TODO: no Endpoints in JS

File: config/PropertyNames.xml [hunk 2]
```diff
@@ -195,363 +195,364 @@ generated from the section label.
...
+        <suffix name="Locator" class="proxy" languages="cpp,csharp,java" />
+        <suffix name="PublishedEndpoints" languages="all" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:236]: PublishedEndpoints is indeed correct for JS. Typically use-case: routing through Glacier2.

File: config/PropertyNames.xml [hunk 3]
```diff
@@ -195,363 +195,364 @@ generated from the section label.
...
+        <property name="Admin.Facets" languages="cpp,csharp,java" />
+        <property name="Admin.InstanceName" languages="cpp,csharp,java" />
+        <property name="Admin.Logger.KeepLogs" languages="cpp,csharp,java" default="100" />
+        <property name="Admin.Logger.KeepTraces" languages="cpp,csharp,java" default="100" />
+        <property name="Admin.Logger.Properties" languages="cpp,csharp,java" />
+        <property name="Admin.ServerId" languages="cpp,csharp,java" />
+        <property name="BackgroundLocatorCacheUpdates" languages="all" default="0" />
+        <property name="BatchAutoFlush" deprecated="true" languages="all" />
+        <property name="BatchAutoFlushSize" default="1024" languages="all" />
+        <property name="ChangeUser" languages="cpp" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:266]: We should consider removing Ice.ChangeUser.

File: config/PropertyNames.xml [hunk 4]
```diff
@@ -195,363 +195,364 @@ generated from the section label.
...
+        <property name="RetryIntervals" languages="all" default="0" />
+        <property name="ServerIdleTime" languages="cpp,csharp,java" default="0" />
+        <property name="SOCKSProxyHost" languages="cpp,csharp,java" />
+        <property name="SOCKSProxyPort" languages="cpp,csharp,java" default="1080" />
+        <property name="StdErr" languages="cpp,csharp,java" />
+        <property name="StdOut" languages="cpp,csharp,java" />
+        <property name="SyslogFacility" languages="cpp,java" default="LOG_USER" />
+        <property name="ThreadPool.Client" class="threadpool" languages="cpp,csharp,java" />
+        <property name="ThreadPool.Server" class="threadpool" languages="cpp,csharp,java" />
+        <property name="ThreadPriority" languages="cpp,csharp,java" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:318]: Remove from cpp?

File: config/PropertyNames.xml [hunk 5]
```diff
@@ -195,363 +195,364 @@ generated from the section label.
...
+        <property name="UDP.SndSize" languages="cpp,csharp,java" />
+        <property name="TCP.Backlog" languages="cpp,csharp,java" />
+        <property name="TCP.RcvSize" languages="cpp,csharp,java" />
+        <property name="TCP.SndSize" languages="cpp,csharp,java" />
+        <property name="UseOSLog" languages="cpp" default="0" />
+        <property name="UseSyslog" languages="cpp,java" default="0" />
+        <property name="UseSystemdJournal" languages="cpp" default="0" />
+        <property name="Warn.AMICallback" languages="cpp,csharp,java" default="1" />
+        <property name="Warn.Connections" languages="all" default="0" />
+        <property name="Warn.Datagrams" languages="all" default="0" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:338]: Warn.Datagrams in JS even though we don't support UDP?

File: config/PropertyNames.xml [hunk 6]
```diff
@@ -195,363 +195,364 @@ generated from the section label.
...
-        <property name="InstanceName" default="IceGridAdmin" />
-        <property name="Server" class="objectadapter" />
-        <property name="Discovery.Address" />
-        <property name="Discovery.Interface" />
-        <property name="Discovery.Lookup" />
-        <property name="Discovery.Reply" class="objectadapter" />
-        <property name="Discovery.Locator" class="objectadapter" />
-        <property name="Trace.Observers" />
-        <property name="Trace.SaveToRegistry" />
+        <property name="AuthenticateUsingSSL" languages="cpp,java" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:400]: I suspect some properties are cpp-only or java-only, e.g. Trace.SaveToRegistry is java-only.

File: config/PropertyNames.xml [hunk 7]
```diff
@@ -195,363 +195,364 @@ generated from the section label.
...
-        <property name="Port" />
+        <property name="TopicManager.[any]" languages="cpp" />
+        <property name="Host" languages="cpp" />
+        <property name="Port" languages="cpp" />
     </section>
 
     <section name="IceBT">
-        <property name="RcvSize" />
-        <property name="SndSize" />
+        <property name="RcvSize" languages="cpp" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:521]: We also have these properties in Java.

File: config/PropertyNames.xml [hunk 8]
```diff
@@ -195,363 +195,364 @@ generated from the section label.
...
+        <property name="LoadOrder" languages="cpp,csharp,java" />
+        <property name="PrintServicesReady" languages="cpp,csharp,java" />
+        <property name="Service.[any]" languages="cpp,csharp,java" />
+        <property name="Trace.ServiceObserver" languages="cpp,csharp,java" />
+        <property name="UseSharedCommunicator.[any]" languages="cpp,csharp,java" />
     </section>
 
     <section name="IceBoxAdmin">
-        <property name="ServiceManager.Proxy" class="proxy"/>
+        <property name="ServiceManager.Proxy" class="proxy" languages="cpp" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:390]: We actually have IceBoxAdmin in java too.

File: config/PropertyNames.xml [hunk 9]
```diff
@@ -195,363 +195,364 @@ generated from the section label.
...
-        <property name="Trace.Observers" />
-        <property name="Trace.SaveToRegistry" />
+        <property name="AuthenticateUsingSSL" languages="cpp,java" />
+        <property name="MetricsConfig" languages="cpp,java" />
+        <property name="Username" languages="cpp,java" />
+        <property name="Password" languages="cpp,java" />
+        <property name="Replica" languages="cpp,java" />
+        <property name="Host" languages="cpp,java" />
+        <property name="Port" languages="cpp,java" />
+        <property name="InstanceName" default="IceGridAdmin" languages="cpp,java" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:407]: This is a strange property with a strange default.

**Review by bernardnormier — APPROVED** [review]

File: config/PropertyNames.xml (hunk 5 above)
Comment (externl) [reply config/PropertyNames.xml:338]: PR bug

**Review by pepone — APPROVED** [review]

**Pull request merged**